### ISSUE TITLE
[feature request] Ability to change animation speed and disable animations entirely

### ISSUE THREAD

**Issue by thejevans** [pop-os/cosmic-comp#376]
Animations when moving windows, adding and removing windows, and changing workspaces are slow and generally unnecessary, even though they are pretty. It would be great if there was a setting in `config.ron` to change the animation durations or set them to 0 to disable them.

**Comment by Drakulix** [comment]
Yes, this is also an accessibility feature. Not the top priority right now, but we will very likely include options to "reduce animations" and to "disable animations" in the future. (Though precisely controlling the animation duration might be out of scope.)

**Comment by vnelis** [comment]
Drakulix please prioritize this accessibility feature! 
Forget about tweaking the animation speed, just allow us to remove the animations altogether. I guess I speak for all the devs in the world here when I say that mapping all my windows to different workspaces is crucial. I personally switch between workspaces every minute (or every few seconds sometimes). These animations are a guaranteed motion sickness. Removing them is literally the first I do in every distro.

**Comment by B83C** [comment]
Please allow disabling animations

**Comment by pspsdev** [comment]
Bumping this up. Having animations while switching workspaces are extremely annoying

**Comment by FinGu** [comment]
+1

**Comment by alvaroaleman** [comment]
Please do not respond with `+1`, it adds literally nothing, just add a thumbs up emoji to the original request. GitHub allows to [sort issues by number of reactions](https://github.com/pop-os/cosmic-comp/issues?q=%20is%3Aissue%20is%3Aopen%20sort%3Areactions-%2B1-desc), sorting by number of comments on the other hand is mostly useless, as there can be a number of reasons for that including a feature being contentious.

**Comment by pudrio** [comment]
+1

**Comment by SeniorMatt** [comment]
+1

**Comment by sudomateo** [comment]
> +1

@SeniorMatt as a fellow Matthew I have to educate you that commenting with "+1" helps no one. Instead, react to the initial message with 👍🏼 that way maintainers can [sort issues by reactions](https://github.com/pop-os/cosmic-comp/issues?q=is%3Aissue%20state%3Aopen%20sort%3Areactions-%2B1-desc).

This was already addressed in https://github.com/pop-os/cosmic-comp/issues/376#issuecomment-3418734444 when someone else commented with a "+1". Please also read the issue in its entirety before commenting.

**Comment by SeniorMatt** [comment]
Oh thank you my fellow Matthew, still learning GitHub a little)

**Comment by sudomateo** [comment]
> Oh thank you my fellow Matthew, still learning GitHub a little)

You're good! I didn't mean anything harshly. We're all here for the right reasons- supporting this project.

**Comment by VictorStankov** [comment]
+1

**Comment by bwks** [comment]
Hey team. Any chance of having this added to `Epoch 2`?

**Comment by mkellyxp** [comment]
100% on this!  I also don't care about controlling the speed of animation, but more just want to disable animations completely.  It looks impressive in a demo, but if you switch around a lot, it triggers motion sickness for sure.

I suspect disabling all animations will also fix my fast alt + tab switching issues as well

**Comment by renan-lopes-eso** [comment]
Im also really looking forward to disable animations, so annoying

**Comment by pspsdev** [comment]
Everyone make sure to thumb up the root issue comment so that it appears at the top of the lists.

**Comment by KaKi87** [comment]
It already appears at the top and it has 55 more upvotes than the next one, so that won't change anything for now.

**Comment by sudomateo** [comment]
God, people on GitHub issues can be really annoying. Reading the useless comments on this issue I've subscribed to is exhausting. I can only imagine how the maintainers must feel.

We get it, you want to disable animations entirely. So do others! We don't need to ask about it every month.

> It already appears at the top and it has 55 more upvotes than the next one, so that won't change anything for now.

It's not only about ranking the issue as first by reactions, but showing just how much interest there is in the issue itself.

**Comment by bwks** [comment]
I wanted to share my workaround which has been helpful to me with the lack of a toggle to turn the animations off. 

If you are not aware, there is a windows stacking feature: https://pop-os.github.io/docs/navigate-pop/tiling-stacking-windows.html#stacking-windows where you can group multiple windows together. You can switch between windows in the stack with {super}+left/right arrow keys. The switching happens with no animations. 

My workflow is:
 - On a tiled workspace
 - open browser
 - press super+s to create a stack
 - open editor (this opens in the stack)
 - open terminal (it also opens in the stack) 

This way I can mostly stay on a single workspace and switch between the apps in the stack without getting motion sickness from the animations. HTH

**Comment by lazylionca** [comment]
I have thumbed the original post, but would also like to add the option to change the animation direction.
Right now the animation moves vertically, but I would prefer it to move horizontal as I have a workspace switcher in my taskbar that is horizontal.

Keep up the great work guys. Loving it so far.

**Comment by leviport** [comment]
@lazylionca have you set the workspaces orientation to horizontal?

https://github.com/user-attachments/assets/2dbf0acf-4c87-4dce-8597-24502b53ec84

**Comment by lazylionca** [comment]
@leviport Doh!

Thanks for the prompt response.

**Comment by csptop1-collab** [comment]
If you're using NixOS you can copy paste the following - it will be equivalent to turning off animations.

Note: This will result in compiling some COSMIC components which can take time depending on your hardware speed.

```nix
nixpkgs.overlays = [ (final: prev: {
  cosmic-session = pkgs.cosmic-session;
  cosmic-greeter = pkgs.cosmic-greeter;
  cosmic-settings = pkgs.cosmic-settings;
  cosmic-launcher = pkgs.cosmic-launcher;
  cosmic-panel = pkgs.cosmic-panel;
  cosmic-applets = pkgs.cosmic-applets;
  cosmic-edit = pkgs.cosmic-edit;
  cosmic-files = pkgs.cosmic-files;
  cosmic-term = pkgs.cosmic-term;
  pop-launcher = pkgs.pop-launcher;
  cosmic-comp = pkgs.cosmic-comp.overrideAttrs (oldAttrs: {
    postPatch = ''
      ${oldAttrs.postPatch or ""}
      
      # Set all animations to 1ms
      find src/shell -type f -name "*.rs" -exec sed -i \
        's/Duration::from_millis([0-9]\+)/Duration::from_millis(1)/g' {} \;
      
      # Fix For Emacs title bar issue.
      sed -i 's/pub const SSD_HEIGHT: i32 = 36;/pub const SSD_HEIGHT: i32 = 0;/g' \
        src/shell/element/window.rs
    '';
  });
})];

services.desktopManager.cosmic.enable = true;
services.system76-scheduler = {
  enable = true;
  package = pkgs.system76-scheduler;
};
```

If you're not using NixOS, you can still use the patch, but you'll have to apply it and compile manually. Maybe consult AI for help.

**Comment by veganomy** [comment]
Just tried `cosmic-session` pacman group to try this desktop environment. Couldn't daily drive due to the lack of animation disabling toggle (since it's giving me motion sickness). So I've went back to gnome, as it allows disabling them.

**Comment by B83C** [comment]
check out niri + dankmaterialshell, usable replacement for now.

**Comment by KaKi87** [comment]
Does this provide the same zero–config mouse-friendly auto-tiling that COSMIC provides ?

**Comment by veganomy** [comment]
Niri is primarily a compositor. Not a desktop environment. Setup everything around it and it'll behave like one.

**Comment by KaKi87** [comment]
So, not zero-config, so, not a replacement.

**Comment by pmiddend** [comment]
> If you're using NixOS you can copy paste the following - it will be equivalent to turning off animations.
> 
> Note: This will result in compiling some COSMIC components which can take time depending on your hardware speed.
> 
> snip
> 
> If you're not using NixOS, you can still use the patch, but you'll have to apply it and compile manually. Maybe consult AI for help.

This actually works! It can be shortened to this though:

```nix
nixpkgs.overlays = [ (final: prev: {
      # Source:
      # https://github.com/pop-os/cosmic-comp/issues/376#issuecomment-3851990669
      cosmic-comp = prev.cosmic-comp.overrideAttrs (oldAttrs: {
        postPatch = ''
          ${oldAttrs.postPatch or ""}

          # Set all animations to 1ms
          find src/shell -type f -name "*.rs" -exec sed -i \
            's/Duration::from_millis([0-9]\+)/Duration::from_millis(1)/g' {} \;

          # Fix For Emacs title bar issue.
          sed -i 's/pub const SSD_HEIGHT: i32 = 36;/pub const SSD_HEIGHT: i32 = 0;/g' \
            src/shell/element/window.rs
        '';
      });
    })]
```

**Comment by Gibbz** [comment]
i find that the 4 finger gesture to switch between desktops if much to fast when i flick the trackpad. Im not usually a fan of animations, but this needs to be slowed down similar to gnome as it causes serious fatigue.... At least a config file to tweak this would be good.
